### PR TITLE
CFE-1098: Rebase upstream v1.35.0

### PR DESCRIPTION
**Pull request by chiragkyal**
**Description of the change:**

Rebase this repo to upstream `v1.35.0` : https://github.com/operator-framework/ansible-operator-plugins/tree/release-v1.35

Steps followed
```
git remote add upstream https://github.com/operator-framework/ansible-operator-plugins.git

git fetch --all

git checkout -b rebase-upstream-v1.35.0

git merge v1.35.0

git push origin rebase-upstream-v1.35.0
```

```
go mod tidy && go mod vendor

git add , commit , push
```

### CODE REVIEW

File: images/ansible-operator/Pipfile [hunk 1]
```diff
@@ -6,9 +6,9 @@ name = "pypi"
 [packages]
 ansible-runner = "~=2.3.3"
 ansible-runner-http = "~=1.0.0"
-ansible-core = "~=2.15.0"
+ansible-core = "~=2.15.9"
 urllib3 = "~=1.26.17"
-kubernetes = "==26.1.0"
+kubernetes = "==29.0.0"
```
Comment (TrilokGeer) [inline images/ansible-operator/Pipfile:11]: Is the version label correct?

Comment (chiragkyal) [reply images/ansible-operator/Pipfile:11]: It's similar to what is present upstream. 

https://github.com/operator-framework/ansible-operator-plugins/blob/release-v1.35/images/ansible-operator/Pipfile#L11

File: Makefile [hunk 2]
```diff
@@ -9,8 +9,8 @@ export IMAGE_VERSION = v1.33.0
 export SIMPLE_VERSION := $(shell (test "$(shell git describe --tags)" = "$(shell git describe --tags --abbrev=0)" && echo $(shell git describe --tags)) || echo $(shell git describe --tags --abbrev=0)+git)
 export GIT_VERSION := $(shell git describe --dirty --tags --always)
 export GIT_COMMIT := $(shell git rev-parse HEAD)
-export K8S_VERSION = 1.27.0
-export ENVTEST_VERSION = 1.27.x
+export K8S_VERSION = 1.28.0
```
Comment (TrilokGeer) [inline Makefile]: A new k8s bump is required upstream for 1.29.

File: images/ansible-operator/Pipfile (hunk 1 above)
Comment (TrilokGeer) [reply images/ansible-operator/Pipfile:11]: It is better to assess if it makes sense to use v30.1.0 based on the release changelog and [compatibilty matrix](https://github.com/kubernetes-client/python/tree/v30.1.0?tab=readme-ov-file#compatibility).

File: Makefile (hunk 2 above)
Comment (chiragkyal) [reply Makefile]: I think this file is Outdated. Upstream is using 1.29 and so as the PR

https://github.com/openshift/ansible-operator-plugins/pull/4/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R12

File: images/ansible-operator/Pipfile (hunk 1 above)
Comment (chiragkyal) [reply images/ansible-operator/Pipfile:11]: I think it should happen upstream first, and we can then pull into downstream.

**Comment by chiragkyal** [comment]
> ci/prow/all 

the `all` can be confusing with running every job. We can rename it.

**Comment by arkadeepsen** [comment]
> > ci/prow/all
> 
> the `all` can be confusing with running every job. We can rename it.

I think we can remove the `ci/prow/all` as it is running `test-static` again. Additionally, we should work on a downstream e2e test using OCP cluster, rather that `test-e2e`.

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved

**Pull request merged**